Added directories `credentials` and `temp`

mlegner opened the pull request:
Store keys and certificates in directory `credentials`, the content of which will be ignored by git, instead of `python`. Also add directory `temp` where the default topology and files generated for users will be stored (instead of `~/scionLabConfigs`), this is also ignored by git.
This PR also fixes a potential problem of a non-standard GOPATH.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/netsec-ethz/scion-coord/56)
<!-- Reviewable:end -->

ercanucan commented:
Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[controllers/api/scion_lab.go, line 299 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssX8lGKWYorn9N9Y24:-KssX8lGKWYorn9N9Y25:bij6p2w) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/controllers/api/scion_lab.go#L299)):*
> ```Go
> func (s *SCIONLabVMController) packageSCIONLabVM(userEmail string) (string, error) {
> 	log.Printf("Packaging SCIONLab VM")
> 	user_package_path := filepath.Join(package_path, userEmail)
> ```

The rationale for creating the configuration in a folder named `scionLabConfigs` was to ensure that we store the tarballs we served to the users in case we need to examine/debug something about it in the future. Putting these into a `temp` folder would be misleading.

---

*[credentials/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssWdBaA18NmzZ0_ttB:-KssWdBaA18NmzZ0_ttC:b-zdekzt) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/credentials/.gitignore#L1)):*
> ```
> # Ignore everything in this directory
> ```

I think we do not need to have separate `.gitignore` files. You can put the folders to be omitted in the top-level `gitignore` file.

---

*[temp/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssXycYRAaOUcOSIbSv:-KssXycZ1aN2N-bCjFUA:b786ge6) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/temp/.gitignore#L1)):*
> ```
> # Ignore everything in this directory
> ```

Same here. We can always use the top-level `.gitignore` file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56)*
<!-- Sent from Reviewable.io -->

mlegner commented:
Addressed all review commands, rebased on current master and made further improvements according to our offline discussion. See details in commit message.

---

Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

*[controllers/api/scion_lab.go, line 253 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-Kssj1gWHGyCF69PVUfK:-Kssj1gZdZ2XceT8i0cY:b-53unxe) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/c8977adbdd75a00cbef51fd9a041b52ca2538fef/controllers/api/scion_lab.go#L253)):*
> python_path+":"+scion_path+":"+scion_web_path

---

*[controllers/api/scion_lab.go, line 299 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssX8lGKWYorn9N9Y24:-KswKWyZlbQB0ePs0GHE:bmd6r7k) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/controllers/api/scion_lab.go#L299)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

The rationale for creating the configuration in a folder named `scionLabConfigs` was to ensure that we store the tarballs we served to the users in case we need to examine/debug something about it in the future. Putting these into a `temp` folder would be misleading.
</blockquote></details>

Reverted these changes.

---

*[credentials/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssWdBaA18NmzZ0_ttB:-KswKbhY76iibFEwsZOm:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/credentials/.gitignore#L1)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

I think we do not need to have separate `.gitignore` files. You can put the folders to be omitted in the top-level `gitignore` file.
</blockquote></details>

Done.

---

*[temp/.gitignore, line 1 at r1](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KssXycYRAaOUcOSIbSv:-KswKcHnVGYqHUCveKAI:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/704a00271d20ca72e7146f69f7b71a6adf879664/temp/.gitignore#L1)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

Same here. We can always use the top-level `.gitignore` file.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-:-KswKecnxvx3BVtdZv6z:bh5iz44)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
Review status: 0 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[.gitignore, line 4 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxF0dP_EOVza6fZK4a:-KsxF-H40pnUVsODSijT:b-suk2n9) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/.gitignore#L4)):*
> ```
> vendor/*/
> temp/
> credentials/
> ```

missing new line.

---

*[main.go, line 34 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxFuwJveYCVk5Xb7W-:-KsxFuwKI_AISP6Q24OE:buseg0g) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/main.go#L34)):*
> ```Go
> 	// check if credential files exist and create necessary directories
> 	for _, f := range []string{api.TrcFile, api.CoreCertFile, api.CoreSigKey} {
> 		if _, err := os.Stat(f); os.IsNotExist(err) {
> ```

What if the error is something else here? The execution will continue?

---

*[main.go, line 35 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxFE61bdf-81YSPGI7:-KsxFE61bdf-81YSPGI8:b-dbf3d4) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/main.go#L35)):*
> ```Go
> 	for _, f := range []string{api.TrcFile, api.CoreCertFile, api.CoreSigKey} {
> 		if _, err := os.Stat(f); os.IsNotExist(err) {
> 			fmt.Println("ERROR: Please make sure that the necessary credential files exist. Consult the README.md for further details.")
> ```

This line seems to be too long.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56)*
<!-- Sent from Reviewable.io -->

mlegner commented:
Review status: 0 of 5 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[.gitignore, line 4 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxF0dP_EOVza6fZK4a:-KsxHMULDtCiJ31QOEdV:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/.gitignore#L4)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

missing new line.
</blockquote></details>

Done.

---

*[main.go, line 34 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxFuwJveYCVk5Xb7W-:-KsxIndvxzFSBhMdrVUd:b-bvj5j2) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/main.go#L34)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

What if the error is something else here? The execution will continue?
</blockquote></details>

Added another conditional clause checking for any error.

---

*[main.go, line 35 at r2](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-KsxFE61bdf-81YSPGI7:-KsxIsKEZV-xi9L47Ina:b-896fix) ([raw file](https://github.com/netsec-ethz/scion-coord/blob/f814f1e27f973bfe6869e1251e5115fa7ce05931/main.go#L35)):*
<details><summary><i>Previously, ercanucan (ercanucan) wrote…</i></summary><blockquote>

This line seems to be too long.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (please rebase)

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56#-:-KsxTQRrOUC8IHgTJXQM:bizmcw8)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
Reviewed 1 of 7 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56)*
<!-- Sent from Reviewable.io -->

ercanucan commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/netsec-ethz/scion-coord/56)*
<!-- Sent from Reviewable.io -->
